contracts/Aavegotchi/ForgeDiamond/facets/ForgeDAOFacet.sol

```diff
@@ -133,35 +135,47 @@ contract ForgeDAOFacet is Modifiers {
-    //        s.geodeWinChance[GODLIKE_RSM] = chances.godlike;
-    //
-    //        emit SetGeodeWinChance(chances);
-    //    }
-    //
-    //    // @notice Allow DAO to set which prizes can be won from a Geode.
-    //    // @param ids Token IDs of the available prizes
-    //    // @param quantities Initial amounts of each prize available
-    //    function setGeodePrizes(uint256[] calldata ids, uint256[] calldata quantities) external onlyDaoOrOwner {
-    //        require(ids.length == quantities.length, "ForgeDAOFacet: mismatched arrays");
-    //
-    //        for (uint256 i; i < ids.length; i++){
-    //            if (s.geodePrizeQuantities[ids[i]] == 0){
-    //                // this ID is deleted from the array in the geode opening function when last item is won.
-    //                s.geodePrizeTokenIds.push(ids[i]);
-    //            }
-    //            s.geodePrizeQuantities[ids[i]] += quantities[i];
-    //        }
-    //
-    //        emit SetGeodePrizes(ids, quantities);
-    //    }
+    // @notice Allow DAO to update percent chance to win from a Geode.
+    // @param points RarityValueIO struct of points
+    function setGeodeWinChanceBips(RarityValueIO calldata chances) external onlyDaoOrOwner {
+        s.geodeWinChanceBips[COMMON_RSM] = chances.common;
+        s.geodeWinChanceBips[UNCOMMON_RSM] = chances.uncommon;
+        s.geodeWinChanceBips[RARE_RSM] = chances.rare;
+        s.geodeWinChanceBips[LEGENDARY_RSM] = chances.legendary;
+        s.geodeWinChanceBips[MYTHICAL_RSM] = chances.mythical;
+        s.geodeWinChanceBips[GODLIKE_RSM] = chances.godlike;
+
+        emit SetGeodeWinChance(chances);
+    }
+
+    // @notice Allow DAO to set which prizes can be won from a Geode.
+    // @param ids Token IDs of the available prizes
+    // @param quantities Initial amounts of each prize available
+    function setGeodePrizes(uint256[] calldata ids, uint256[] calldata quantities) external onlyDaoOrOwner {
+        require(ids.length == quantities.length, "ForgeDAOFacet: mismatched arrays");
+
+        for (uint256 i; i < s.geodePrizeTokenIds.length; i++) {
+            delete s.geodePrizeQuantities[s.geodePrizeTokenIds[i]];
+        }
+        delete s.geodePrizeTokenIds;
+        for (uint256 i; i < ids.length; i++) {
+            if (s.geodePrizeQuantities[ids[i]] == 0) {
+                // this ID is deleted from the array in the geode opening function when last item is won.
+                s.geodePrizeTokenIds.push(ids[i]);
+            }
+            s.geodePrizeQuantities[ids[i]] = quantities[i];
+        }
+
+        emit SetGeodePrizes(ids, quantities);
+    }
+
+    function getGeodePrizesRemaining() external view returns (uint256[] memory, uint256[] memory) {
+        uint256[] memory quantities = new uint256[](s.geodePrizeTokenIds.length);
+        for (uint256 i; i < s.geodePrizeTokenIds.length; i++) {
+            quantities[i] = s.geodePrizeQuantities[s.geodePrizeTokenIds[i]];
```

Comment:
I'm not totally sure you need the `geodePrizeQuantities` storage variable. You could just use the `geodePrizeTokenIds` array and then track the balance of each tokenId in the contract. 

The issue with using "virtual" supply is that there could be discrepancies between the supply that the contract thinks it has (geodePrizeQuantities) and the amount actually in the contract (the token balance). Would be cleaner to use the token balance itself I think

Reply:
I prefer to use a "virtual" supply to keep tighter control over what is a geode prize instead of whatever is sent to the contract.